src/g_editor.c

```diff
@@ -1449,7 +1447,8 @@ int canvas_undo_canvas_apply(t_canvas *x, void *z, int action)
             glist_noselect(x);
             gobj_vis(&x->gl_gobj, x->gl_owner, 0);
             gobj_vis(&x->gl_gobj, x->gl_owner, 1);
-            canvas_redraw(x->gl_owner);
+            if (x->gl_havewindow)
```

Comment:
shouldn't this be `x->gl_owner->gl_havewindow`?

Reply:
yes. 2c7b4c1937c0e57d24d48de3d83969004e080bab